### PR TITLE
Google Analytics events for attribution button click

### LINKED ISSUE #187: add attribution button to google analytics
so clicks of both "copy to text" and "copy to html" show up in Google analytics

### PR DESCRIPTION
**Pull request by brenoferreira**
Sends events to Google Analytics when the user clicks the attribution button

Fixes #187

### CODE REVIEW

**Comment by brenoferreira** [comment]
screen shot 2019-02-25 at 16 50 16
_screenshot of attribution events on GA_

File: src/analytics/GoogleAnalytics.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+  if (!gaAvailable) {
+    console.warn('Google Analytics is unavailable. Unable to send any events.');
+  }
+
+  return gaAvailable && !doNotTrackEnabled();
+}
+
+function sendEvent(event) {
+  console.log('Sending GA event: ', event);
```
Comment (kgodey) [inline src/analytics/GoogleAnalytics.js]: this looks like a debug log, does it make sense to ship to production?

File: src/analytics/GoogleAnalytics.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+function setCurrentPage(page) {
+  set('page', page);
+}
+
+const GoogleAnalytics = () => {
+  const enabled = isTrackingEnabled();
+  return {
+    anonymizeIpAddress() {
+      if (enabled) {
+        set('anonymizeIp', true);
```
Comment (kgodey) [inline src/analytics/GoogleAnalytics.js]: I think we should always anonymize IPs.

File: src/analytics/events.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import { Event } from './GoogleAnalytics';
+
+export function CopyTextAttribution(text) {
+  return new Event('Attribution', 'Copy Text', text);
```
Comment (kgodey) [inline src/analytics/events.js]: could we log the anonymized IP here?

File: src/analytics/GoogleAnalytics.js (hunk 2 above)
Comment (brenoferreira) [reply src/analytics/GoogleAnalytics.js]: Done https://github.com/creativecommons/cccatalog-frontend/pull/190/commits/2a6a9cf16a9a2eeb3639402bf76032e14e75089e

File: src/analytics/events.js (hunk 3 above)
Comment (brenoferreira) [reply src/analytics/events.js]: Why log the anonymized IP?

Comment (kgodey) [reply src/analytics/events.js]: To keep track of how many users are using the copy attribution feature (is one user using it a lot or is it a bunch of different users?). Google Analytics might support this out of the box though, so I'm not sure if we need to log it here.

Comment (brenoferreira) [reply src/analytics/events.js]: I'm not sure whether events are individualized or not on GA. My first guess would be that they're not.

**Comment by kgodey** [comment]
Decided on standup that this will be merged as-is and we'll create a separate issue to research how to use Google Analytics to figure out the number of distinct users that are clicking the attribution buttons.

**Comment by janeatcc** [comment]
In this separate issue, let's make sure to involve @sarahpearson and have that sign-off before implementation

**Pull request merged**